service/src/main.rs

```diff
@@ -1,229 +1,14 @@
-            .map_or(false, |value| {
-                value.to_str().map(|value| value == "true").unwrap_or(false)
-            });
-
-        let body = response
-            .text()
-            .await
-            .map_err(SubgraphServiceError::QueryForwardingError)?;
-
-        Ok((request, SubgraphServiceResponse::new(body, attestable)))
-    }
-}
-
-/// Run the subgraph indexer service
 #[tokio::main]
-async fn main() -> Result<(), Error> {
-    tracing_subscriber::fmt::init();
-
-    // Parse command line and environment arguments
-    let cli = Cli::parse();
-
-    // Load the json-rpc service configuration, which is a combination of the
-    // general configuration options for any indexer service and specific
-    // options added for JSON-RPC
-    let config = match Config::load(&cli.config) {
-        Ok(config) => config,
-        Err(e) => {
-            error!(
-                "Invalid configuration file `{}`: {}",
-                cli.config.display(),
-                e
-            );
-            std::process::exit(1);
-        }
-    };
-
-    // Parse basic configurations
-    build_info::build_info!(fn build_info);
-    let release = IndexerServiceRelease::from(build_info());
-
-    // Some of the subgrpah service configuration goes into the so-called
-    // "state", which will be passed to any request handler, middleware etc.
-    // that is involved in serving requests
-    let state = Arc::new(SubgraphServiceState {
-        config: config.clone(),
-        database: database::connect(&config.common.database.postgres_url).await,
-        cost_schema: routes::cost::build_schema().await,
-        graph_node_client: reqwest::ClientBuilder::new()
-            .tcp_nodelay(true)
-            .timeout(Duration::from_secs(30))
-            .build()
-            .expect("Failed to init HTTP client for Graph Node"),
-        graph_node_status_url: config
-            .common
-            .graph_node
-            .as_ref()
-            .expect("Config must have `common.graph_node.status_url` set")
-            .status_url
-            .clone(),
-        graph_node_query_base_url: config
-            .common
-            .graph_node
-            .as_ref()
-            .expect("config must have `common.graph_node.query_url` set")
-            .query_base_url
-            .clone(),
-    });
-
-    IndexerService::run(IndexerServiceOptions {
-        release,
-        config: config.common.clone(),
-        url_namespace: "subgraphs",
-        metrics_prefix: "subgraph",
-        service_impl: SubgraphService::new(state.clone()),
-        extra_routes: Router::new()
-            .route("/cost", post(routes::cost::cost))
-            .route("/status", post(routes::status))
-            .with_state(state),
-    })
-    .await
+async fn main() {
+    if let Err(e) = run().await {
+        eprintln!("Indexer service error: {e}");
+        // https://doc.rust-lang.org/std/process/fn.exit.html
+        drop(e);
```

Comment:
Why the explicit drop? The program is going to be terminated anyway.

Reply:
The idea comes from the docs that I've linked to there, namely:

>If a clean shutdown is needed it is recommended to only call this function at a known point where there are no more destructors left to run.

Reply:
I see. That lead me to some googling and I found a solution that I think is even more elegant here considering the post-refactor state of things: https://doc.rust-lang.org/std/process/struct.ExitCode.html

Reply:
Well, maybe I find it more elegant because it looks more like what we'd do in C, and C feels cozy to me :stuck_out_tongue: